### ISSUE TITLE
PIXIO Implementation Plan

### ISSUE THREAD

**Issue by gabrielfruet** [lightly-ai/lightly#1894]
PIXIO differs from standard MAE in three key differences:

1. Deeper decoder
2. More mask granularity
3. Increase of CLS tokens

Research repo is hosted here:

[facebookresearch/pixio](https://github.com/facebookresearch/pixio)

## 1. Deeper decoder

MAE Decoder generally has 8 blocks due to their philosophy of letting the encoder do the hard work. Pixio drifts from this by explictly stating that the encoder is responsible for encoding semantics and the decoder is responsible for caring about the pixel-level detail. By separating this they create a much richer latent space for the encoder.

[Decoder forward of Pixio](https://github.com/facebookresearch/pixio/blob/b83555d6581f4906f6f1fc1ae0abb04f622db32f/pretraining/models_pixio.py#L222-L256)

[Our MAEDecoder](https://github.com/lightly-ai/lightly/blob/ee30cd481d68862c80de4ef45920cfe1ab1f67b1/lightly/models/modules/masked_autoencoder.py#L380C1-L506C32)

This could be a module, just like we have the MAEDecoder. We could likely subclass from it also.

## 2. More mask granularity

They use a mask that is not simply randomly dropping points, they mask by blocks.

![](https://github.com/facebookresearch/pixio/raw/main/assets/pixio.png)

We also already have that implemented

[Block Masking](https://github.com/lightly-ai/lightly/blob/ee30cd481d68862c80de4ef45920cfe1ab1f67b1/lightly/models/utils.py#L654-L747)

## 3. Increase of CLS tokens

An example for `timm` models is to increase the number of `reg_tokens`

By default it has only 1 CLS token, but we can add more via REG tokens.

[VisionTransformer from timm](https://github.com/huggingface/pytorch-image-models/blob/56467ad5f2b443d454df1a5bb8b23c04697823c7/timm/models/vision_transformer.py#L807-L808)

-------

Let me know your thoughts.
